### ISSUE TITLE
Fail the build when javadoc.options can't map characters (e.g. when printing emoji to a windows-1252 file)

### ISSUE THREAD

**Issue by vlsi** [gradle/gradle#22575]
### Expected Behavior

Gradle should fail the build when the configured encoding does not support some of the characters.
For instance, if user puts characters that happen to be unmappable to `javadoc.options`, then Gradle should fail rather than transparently replace them.

The unmappable characters can happen in both user-provided options (e.g. javadoc footer/header), and in file paths (e.g. project directory can have non-ASCII chars).

It might be that the issue has a broader scope than just `javadoc`, however, I believe `javadoc` failures are very common symptoms of unmappable char issues.

### Current Behavior

Currently, Gradle happily produces a corrupted file which causes issues down the line.

For instance, if I use `windows-1252` encoding, and I configure

```kotlin
tasks.withType<Javadoc>().configureEach {
    (options as StandardJavadocDocletOptions).apply {
        bottom = "©"
    }
}
```

then Gradle produces options file with unmappable character which literally corrupts user data:

```
-bottom '�'
```

In the context of javadoc, it causes weird "javadoc: error - cannot read Input length = 1" which are far from easy for the analysis for the regular user.

### Context

* https://github.com/gradle/gradle/issues/22574

**Comment by ov7a** [comment]
Sorry for the late reply.

While this feature request is reasonable from a user perspective, it might be hard to resolve it from a technical perspective completely.

**Comment by vlsi** [comment]
@ov7a , Could you please clarify what is hard from the technical perspective?

Let me rephrase the suggestion:
* Currently Gradle generates `javadoc.options` file. The suggestion is that Gradle should FAIL the build in case the file contains unmappable character.

In other words, this exact test must throw exception rather than asert `? ? ?`: https://github.com/gradle/gradle/blob/46c40f94cad9327d502f06256c24cd9196859d1c/platforms/jvm/language-java/src/integTest/groovy/org/gradle/api/tasks/javadoc/JavadocFileEncodingIntegrationTest.groovy#L47

Java supports `java.nio.charset.CodingErrorAction#REPORT` since Java 1.4, so detecting and reporting the unmappable characters should not be a problem.

WDYT?

**Comment by ov7a** [comment]
@vlsi I meant that in a broader scope, encoding detection is hard since it's just bytes. I was thinking of examples like KOI8-R <-> CP1251 - it's hard to reliably detect encoding issues there.

But your point on `java.nio.charset.CodingErrorAction#REPORT` is completely on spot, thanks for sharing the additional context!
